src/crunch.js

```diff
@@ -26,8 +26,10 @@ function insert(value, index, values) {
 }
 
 module.exports = function crunch(data) {
-  const index = new Map();
   const values = [];
-  flatten(data, index, values);
+  if (typeof data !== 'undefined') {
+    const index = new Map();
+    flatten(data, index, values);
+  }
```

Comment:
We prefer having [guards](https://en.wikipedia.org/wiki/Guard_(computer_science)) specified at the start of the function as a precondition with an early return. Sorry we don't have a style guide on this project. No way you'd have known.

Something like
```js
function crunch(data) {
  if (isUndefined(data)) {
    return [];
  }
  ...
}
```

Would you mind changing this to conform to that pattern? (Adding an isUndefined util would be 👍 too)